caas/kubernetes/provider/events.go

```diff
@@ -59,17 +60,14 @@ func (k *kubernetesClient) getEvents(objName string, objKind string) ([]core.Eve
 }
 
 func (k *kubernetesClient) watchEvents(objName string, objKind string) (watcher.NotifyWatcher, error) {
-	selector := fields.AndSelectors(
-		fields.OneTermEqualSelector("involvedObject.name", objName),
-		fields.OneTermEqualSelector("involvedObject.kind", objKind),
-	).String()
-	events := k.client().CoreV1().Events(k.namespace)
-	w, err := events.Watch(v1.ListOptions{
-		FieldSelector: selector,
-		Watch:         true,
-	})
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return k.newWatcher(w, objName, k.clock)
+	factory := informers.NewSharedInformerFactoryWithOptions(k.client(), 0,
```

Comment:
worth to wrap this to a factoryGetter like 
```go
func (k *kubernetesClient) getInformerFactory(selectors ...Selector) SharedInformerFactory {}
```

Reply:
I don't see the needs. Whats the gain here besides indirection?